src/scala/com/github/johnynek/bazel_deps/CoursierResolver.scala

```diff
@@ -51,57 +51,96 @@ class CoursierResolver(servers: List[MavenServer], ec: ExecutionContext, runTime
+  def getShas(m: List[MavenCoordinate]): Task[SortedMap[MavenCoordinate, ResolvedShasValue]] = {
 
     type L[x] = ValidatedNel[String, x]
     type N[x] = Nested[Task, L, x]
 
-    def lookup(c: MavenCoordinate): N[ResolvedSha1Value] = {
+    def lookup(c: MavenCoordinate): N[ResolvedShasValue] = {
 
-      def downloadSha1s(as: List[coursier.Artifact]): Task[Option[ResolvedSha1Value]] =
-        as.foldM(Option.empty[ResolvedSha1Value]) {
+      def downloadShas(digestType: DigestType, as: List[coursier.Artifact]): Task[Option[ShaValue]] =
+        as.foldM(Option.empty[ShaValue]) {
           case (s @ Some(r), _) => Task.point(s)
-          case (None, a) => downloadSha1(a)
+          case (None, a) => downloadSha(digestType, a)
         }
 
-      def downloadSha1(a: coursier.Artifact): Task[Option[ResolvedSha1Value]] =
+      def downloadSha(digestType: DigestType, a: coursier.Artifact): Task[Option[ShaValue]] =
         Cache.file[Task](a).run.map {
           case Left(error) =>
             logger.info(s"failure to download ${a.url}, ${error.describe}")
             None
           case Right(file) =>
             val serverId = serverFor(a).fold("")(_.id)
-            val o = Sha1Value.parseFile(file).map(s => ResolvedSha1Value(s, serverId)).toOption
+            val o = ShaValue.parseFile(digestType, file).toOption
             o.foreach { r =>
-              logger.info(s"SHA-1 for ${c.asString} downloaded from ${a.url} (${r.sha1Value.toHex})")
+              logger.info(s"SHA-1 for ${c.asString} downloaded from ${a.url} (${r.toHex})")
             }
             o
         }
 
-      def computeSha1(artifact: coursier.Artifact): Task[ResolvedSha1Value] =
+      def computeSha(digestType: DigestType, artifact: coursier.Artifact): Task[ShaValue] =
         Cache.file[Task](artifact).run.flatMap { e =>
           resolverMonad.fromTry(e match {
             case Left(error) =>
               Failure(FileErrorException(error))
             case Right(file) =>
-              Sha1Value.computeShaOf(file).map { sha =>
-                logger.info(s"SHA-1 for ${c.asString} computed from ${artifact.url} (${sha.toHex})")
-                val serverId = serverFor(artifact).fold("")(_.id)
-                ResolvedSha1Value(sha, serverId)
-              }
+              ShaValue.computeShaOf(digestType, file)
           })
         }
 
-      def computeSha1s(as: List[coursier.Artifact]): Task[ResolvedSha1Value] =
+      def computeShas(digestType: DigestType, as: List[coursier.Artifact], errorMessage: Option[String] = None): Task[ShaValue] =
         as match {
-          case Nil => resolverMonad.raiseError(new RuntimeException(s"we could not download the artifact ${c.asString} to compute the hash"))
+          case Nil => resolverMonad.raiseError(new RuntimeException(s"we could not download the artifact ${c.asString} to compute the hash for digest type ${digestType} with error ${errorMessage.get}"))
```

Comment:
this will throw is `errorMessage` is None, right? maybe better to take a cats.data.NonEmptyList for `as` and when you get down to the last just put the final error in.